### PR TITLE
feat: add time based seed to random.h

### PR DESCRIPTION
**Pull request by AvikantSrivastava**
## Proposed changes
The seed used in the random module was fixed to 0. This PR uses a time based seed for initialization

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Comment by AvikantSrivastava** [comment]
@awni can you review this PR?

File: mlx/random.h [hunk 1]
```diff
@@ -18,12 +19,16 @@ class KeySequence {
 
   // static default
   static KeySequence& default_() {
-    static KeySequence ks(0);
+    static KeySequence ks(getCurrentTimeSeed());
```
Comment (LeonEricsson) [inline mlx/random.h:22]: get_current_time_seed()

Comment (AvikantSrivastava) [reply mlx/random.h]: done

**Review by awni — COMMENTED** [review]
Looks good to me. I would check that the tests pass to be sure we weren't relying on a the PRNG default anywhere.

**Comment by AvikantSrivastava** [comment]
> Looks good to me. I would check that the tests pass to be sure we weren't relying on a the PRNG default anywhere.

I have not added unittests for this particular change yet, should I add them?

I ran the existing test suite, all the test cases are passing

**Review by awni — APPROVED** [review]
Thanks for the contribution, LGTM!

**Pull request merged**